impl/src/main/java/io/kroxylicious/testing/kafka/testcontainers/TestcontainersKafkaCluster.java

```diff
@@ -577,29 +583,46 @@ private InspectContainerCmd buildInspectionCommandFor(KafkaContainer kc) {
         return kc.getDockerClient().inspectContainerCmd(kc.getContainerId());
     }
 
-    @SuppressWarnings({ "try" })
-    private static String createNamedVolume() {
-        var volumeCmd = DockerClientFactory.lazyClient().createVolumeCmd();
+    @SuppressWarnings({ "try", "resource" })
+    private static void createNamedVolume(String volumeName, String testName) {
+        pokeContainerEngine();
```

Comment:
I think this logic should be put in front of the other `lazyClient` calls too as any of them could fail the same way if 5 seconds have elapsed and the podman process behind the systemd socket has stopped.

So maybe introduce a method so all the other usages would change from `DockerClientFactory.lazyClient` to:

```
@NotNull
    private static DockerClient getPreparedDockerClient() {
        DockerClient dockerClient = DockerClientFactory.lazyClient();
        pokeContainerEngine(dockerClient);
        return dockerClient;
    }
    ```

Reply:
Yeah my intention was something very similar once I had validates this worked.